pkg/controller/mysqlbackup/internal/syncer/job.go

```diff
@@ -109,9 +109,10 @@ func (s *jobSyncer) getBackupCandidate() string {
 			return node.Name
 		}
 	}
-	log.Info("no healthy slave node found so returns the master node", "default_node", s.cluster.GetPodHostname(0),
+
+	log.Info("no healthy slave node found so returns the master node", "default_node", s.cluster.GetMasterHost(),
```

Comment:
Changing the key from `default_node` to `master_node` would be better.

The `default_node` key in log message should be `s.cluster.GetPodHostname(0)` because `s.cluster.GetMasterHost()` will choose the pod-0 as default node if no master is found. 

https://github.com/presslabs/mysql-operator/blob/fcbb5c30cd28d2be858f9601306eac71510baabf/pkg/internal/mysqlcluster/mysqlcluster.go#L163

Note:
Maybe this default backup candidate route should also be fixed in other issue because this default route indicates that all master and slaves might be broken.

Reply:
> Changing the key from default_node to master_node would be better.

That makes sense to me, I'll update that today.

> The default_node key in log message should be s.cluster.GetPodHostname(0)

I'm not sure if you mean that we should be logging `s.cluster.GetPodHostname(0)` [here](https://github.com/dougfales/mysql-operator/blob/fallback-to-current-master-pod-for-backups/pkg/controller/mysqlbackup/internal/syncer/job.go#L113) instead of `s.cluster.GetMasterHost()`? I think the log message is referring to the value we will return, so I think it should be `s.cluster.GetMasterHost()` since that's what we're returning.

> Maybe this default backup candidate route should also be fixed in other issue because this default route indicates that all master and slaves might be broken.

Yes, I think there is a more general problem here of what do we do when 1) no replicas are available **and** 2) we do not find a master? Returning a default pod is not as good as returning some kind of error condition so we can report that no viable backup sources were found. 

I do think that should be another issue, and I think your solution of just returning `s.cluster.GetMasterHost()` to address this issue is a good next step for now because it means a backup can still be taken.

Reply:
> I'm not sure if you mean that we should be logging s.cluster.GetPodHostname(0) here instead of s.cluster.GetMasterHost()? 

No, not at all.

I just meant if you won't change the key `default_node`, the value `s.cluster.GetPodHostname(0)` would be more appropriate. Sorry for my confusing statement.

> I do think that should be another issue, and I think your solution of just returning s.cluster.GetMasterHost() to address this issue is a good next step for now because it means a backup can still be taken.

I totally agree.

Reply:
Thanks for clarifying @yoichiwo7! And thanks for raising this issue in the first place.